query_optimizer/ExecutionGenerator.cpp

```diff
@@ -1527,5 +1527,11 @@ void ExecutionGenerator::convertTableGenerator(
   temporary_relation_info_vec_.emplace_back(tablegen_index, output_relation);
 }
 
+S::InsertDestination* ExecutionGenerator::addNewInsertDestinationToQueryContext() {
+  std::unique_ptr<S::InsertDestination> insert_destination_proto(query_context_proto_->add_insert_destinations());
```

Comment:
Because that makes it easier to pass on the pointer from this function at the call site. Do you have any alternative in mind?